Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp

```diff
@@ -398,17 +398,15 @@ int32_t LibWebRTCCodecs::releaseDecoder(Decoder& decoder)
 }
 
 // May be called on any thread.
-void LibWebRTCCodecs::flushDecoder(Decoder& decoder, Function<void()>&& callback)
+Ref<GenericPromise> LibWebRTCCodecs::flushDecoder(Decoder& decoder)
 {
     Locker locker { m_connectionLock };
-    if (!decoder.connection || decoder.hasError) {
-        callback();
-        return;
-    }
+    if (!decoder.connection || decoder.hasError)
+        return GenericPromise::createAndResolve();
 
-    decoder.connection->send(Messages::LibWebRTCCodecsProxy::FlushDecoder { decoder.identifier }, 0);
-    Locker flushLocker { decoder.flushCallbacksLock };
-    decoder.flushCallbacks.append(WTFMove(callback));
+    return decoder.connection->sendWithPromisedReply(Messages::LibWebRTCCodecsProxy::FlushDecoder { decoder.identifier }, 0)->whenSettled(workQueue(), [] (auto&&) {
```

Comment:
I think we should be able to directly return the promise with the right PromiseConverter, removing the need to dispatch another task.

Reply:
I have that in another patch, and will do in a follow-up as it requires some changes to Connection.h.
I want to remove all these whenSettled at once to remove some potential race conditions between decode/flush.